documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,299 @@
+  }
+
+  @POST
+  public void add(FruitDto fruit) {
+    fruitService.save(covertFromDto(fruit));
+  }
+
+  private Fruit covertFromDto(FruitDto fruitDto) {
+    return new Fruit(fruitDto.getName(), fruitDto.getDescription(), STORE_NAME);
+  }
+}
+----
+
+The `list` and `add` operations are executed for the `STORE_NAME` id. This is a partition key of the data model.
+We can easily retrieve all rows from cassandra using the partition key. They will be sorted by the clustering key.
+`FruitResource` is using `FruitService` which encapsulates the data access logic.
+
+When creating the REST API we should not share the same entity object between REST API and data access layers.
+They should not be coupled to allow the API to evolve independently of the storage layer.
+This is the reason why the API is using a `FruitDto` class.
+This class will be used by Quarkus to convert JSON to java object for client request and java object to JSON for the response.
+The translation is done by quarkus-resteasy extension.
+[source,java]
+----
+public class FruitDto {
+
+  private String name;
+  private String description;
+
+  public FruitDto() {}
+
+  public FruitDto(String name, String description) {
+    this.name = name;
+    this.description = description;
+  }
+  // getters and setters omitted for brevity
+}
+----
+
+One important thing to note is that having a default constructor is required by the JSON serialization layer.
+
+== Configuring the Cassandra database
+The main two properties to configure are `contact-points`, to access the Cassandra database, and `local-datacenter`, which is required by the driver.
+
+A sample configuration should look like this:
+
+[source,properties]
+----
+# configure the Cassandra client for a replica set of two nodes
+quarkus.cassandra.contact-points={cassandra_ip}:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter={dc_name}
+----
+
+In this example, we are using a single instance running on localhost:
+
+[source,properties]
+----
+# configure the Cassandra client for a single instance on localhost
+quarkus.cassandra.contact-points=127.0.0.1:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter=datacenter1
+----
+
+You can configure all Java driver settings using `application.conf` or `application.json` files.
+They will be passed automatically to the underlying driver configuration mechanism.
+Settings defined in `application.properties` with the `quarkus.cassandra` prefix will have priority over settings defined in `application.conf` or `application.json`.
+
+To see the full list of settings, please refer to the {reference-conf-url}[reference.conf].
+
+== Running a Cassandra Database
+By default, `CassandraClient` is configured to access a local Cassandra database on port 9042 (the default Cassandra port).
+If you have a local running database on this port, you should assert that:
+`quarkus.cassandra.load-balancing-policy.local-datacenter` setting is equal to dc of your local cassandra instance.
```

Comment:
Maybe we should specify that the value can be found with `SELECT data_center FROM system.local`, that might not be obvious for users not familiar with Cassandra.